src/daemon.rs

```diff
@@ -9,11 +12,13 @@ use std::{
     path::Path,
     process::{exit, id},
     thread::sleep,
-    time::Duration,
     time::SystemTime,
 };
 use sysinfo::{ProcessExt, System, SystemExt};
 
+use signal_hook::consts::signal::*;
+use signal_hook::flag as signal_flag;
```

Comment:
https://github.com/waycrate/swhkd/pull/55 switched to tokio with inbuilt signal handler which is able to break out of tokios event loop. Hence this pull request will have to be closed.

I'm really sorry.